src/dex/ekubo/ekubo.ts

```diff
@@ -0,0 +1,657 @@
+
+  private interval?: NodeJS.Timeout;
+
+  // Caches the number of decimals for TVL computation purposes
+  private readonly decimals: Record<string, number> = {
+    [ETHER_ADDRESS]: 18,
+  };
+
+  public constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+  ) {
+    super(dexHelper, dexKey);
+
+    this.logger = dexHelper.getLogger(dexKey);
+    this.config = EkuboConfig[dexKey][network];
+    this.core = new Contract(this.config.core, CoreABI, dexHelper.provider);
+    this.coreIface = new Interface(CoreABI);
+    this.dataFetcher = new Contract(
+      this.config.dataFetcher,
+      DataFetcherABI,
+      dexHelper.provider,
+    );
+    this.routerIface = new Interface(RouterABI);
+
+    // 0 are vanilla pools
+    this.supportedExtensions = [0n, BigInt(this.config.oracle)];
+  }
+
+  // Periodically schedules fetching pool keys from the Ekubo API and filling in details with the quote data fetcher
+  public async initializePricing(blockNumber: number) {
+    await this.updatePoolMap(blockNumber);
+
+    this.interval = setInterval(async () => {
+      await this.updatePoolMap(await this.dexHelper.provider.getBlockNumber());
+    }, POOL_MAP_UPDATE_INTERVAL_MS);
+  }
+
+  // LEGACY
+  public getAdapters(
+    _side: SwapSide,
+  ): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  public async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    _side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const [token0, token1] = sortAndConvertTokens(srcToken, destToken);
+    const pair = hexStringTokenPair(token0, token1);
+
+    let poolKeys: PoolKey[];
+    try {
+      poolKeys = (await this.fetchAllPoolKeys()).filter(
+        poolKey => poolKey.token0 === token0 && poolKey.token1 === token1,
+      );
+    } catch (err) {
+      this.logger.error(
+        `Fetching pools from Ekubo API for token pair ${pair} failed, falling back to default pool parameters: ${err}`,
+      );
+
+      poolKeys = FALLBACK_POOL_PARAMETERS.map(
+        params =>
+          new PoolKey(
+            token0,
+            token1,
+            new PoolConfig(params.tickSpacing, params.fee, 0n),
+          ),
+      );
+
+      if ([token0, token1].includes(NATIVE_TOKEN_ADDRESS)) {
+        poolKeys.push(
+          new PoolKey(
+            token0,
+            token1,
+            new PoolConfig(
+              FULL_RANGE_TICK_SPACING,
+              0n,
+              BigInt(this.config.oracle),
+            ),
+          ),
+        );
+      }
+    }
+
+    const uninitializedPoolKeys: PoolKey[] = [];
+    const initializedPoolKeys: PoolKey[] = [];
+
+    for (const poolKey of poolKeys) {
+      (this.pools.has(poolKey.string_id)
+        ? initializedPoolKeys
+        : uninitializedPoolKeys
+      ).push(poolKey);
+    }
+
+    const promises = this.initializePools(uninitializedPoolKeys, blockNumber);
```

Comment:
no need to initialize pools here, we can do it in `updatePoolMap` only

Reply:
removed initialization here